### PR TITLE
Add Service Guidelines

### LINKED ISSUE #417: Document initialization process for services
@gravityfox suggested that we standardize the initialization process for services in general and especially for the EconomyAPI. This includes the init, pre-init and post-init events.

### PR DESCRIPTION
**Pull request by Flibio**
Adds service guidelines to the services page. Closes #417.

Let me know if the if the information I used is inaccurate.

### CODE REVIEW

**Comment by Tzky** [comment]
You might want to link to https://docs.spongepowered.org/en/plugin/lifecycle.html

**Comment by Flibio** [comment]
@Tzky Done.

**Comment by saladoc** [comment]
@gabizou should probably take a look at this since we're basically describing a contract here.

Important thing: During the `GameInitializationEvent` plugins that want to use the service may determine if it is / will be provided and may store a reference if they want to keep one.

**Comment by Flibio** [comment]
@saladoc Are we waiting for response from gabizou?

**Comment by saladoc** [comment]
No we're not. I've seen a statement on when services should be ready, but I can't seem to find it. I do remember however, that being ready during the Initialization Phase was something that cannot be achieved.
After rereading the lifecycle docs as well I think Services should be registered by the `GamePostInitializationEvent` (Inter-plugin Communication should be ready to occur, Plugins providing an API accept baic requests). And fully operational by `GameAboutToStartServerEvent`

File: source/plugin/services.rst [hunk 1]
```diff
@@ -11,8 +11,21 @@ service manager:
 
 If you need to get an object reference to something, just get it off the service manager.
 
+Service Guidelines
+==================
+
+* Services should be registered during the ``POST_INITIALIZATION`` game state.
```
Comment (saladoc) [inline source/plugin/services.rst]: at the latest

**Pull request merged**